ambari-server/src/main/java/org/apache/ambari/server/controller/AmbariManagementControllerImpl.java

```diff
@@ -2767,31 +2767,34 @@ private void createHostAction(Cluster cluster,
     List<OsSpecific.Package> packages =
         getPackagesForStackServiceHost(ambariMetaInfo.getStack(stackId), serviceInfo, hostParams, osFamily);
     String packageList = gson.toJson(packages);
-    hostParams.put(PACKAGE_LIST, packageList);
+    commandParams.put(PACKAGE_LIST, packageList);
 
     Set<PropertyInfo> stackProperties = ambariMetaInfo.getStackProperties(stackInfo.getName(), stackInfo.getVersion());
 
-    Set<String> userSet = configHelper.getPropertyValuesWithPropertyType(PropertyType.USER, cluster, clusterDesiredConfigs, servicesMap, stackProperties);
+    Set<PropertyInfo> clusterProperties = ambariMetaInfo.getClusterProperties();
+
+    Set<String> userSet = configHelper.getPropertyValuesWithPropertyType(PropertyType.USER, cluster, clusterDesiredConfigs, servicesMap, stackProperties, clusterProperties);
     String userList = gson.toJson(userSet);
     hostParams.put(USER_LIST, userList);
 
     //Create a user_group mapping and send it as part of the hostLevelParams
     Map<String, Set<String>> userGroupsMap = configHelper.createUserGroupsMap(
-      cluster, clusterDesiredConfigs, servicesMap, stackProperties);
+      cluster, clusterDesiredConfigs, servicesMap, stackProperties, clusterProperties);
     String userGroups = gson.toJson(userGroupsMap);
     hostParams.put(USER_GROUPS, userGroups);
```

Comment:
I think this is deadcode

getMetadataClusterLevelParams() is what creates the clusterLevelParams. I don't know where we use these hostParams anymore since these properties don't land in command.json as "hostLevelParams". 


Can you check if we need to have the code around hostParams in createHostAction() anymore. Also please look at getMetadataClusterLevelParams that might need fixing

Reply:
trunk has hostParams.put(USER_LIST, userList) hence I did not remove it.
getMetadataClusterLevelParams makes a call to configHelper.getPropertyValuesWithPropertyType which already has the fix for searching in clusterSettings.